src/controllers/emailVault/emailVault.ts

```diff
@@ -306,11 +306,19 @@ export class EmailVaultController extends EventEmitter {
 
     let result: Boolean | null = false
     let magicKey = await this.#getMagicLinkKey(email)
+
     if (!magicKey?.key && !this.#shouldStopConfirmationPolling) {
       await this.handleMagicLinkKey(email, async () => {
         magicKey = await this.#getMagicLinkKey(email)
       })
     }
+
+    if (this.#shouldStopConfirmationPolling) {
+      this.#isUploadingSecret = false
+      this.emitUpdate()
+      return
+    }
```

Comment:
Why don't we put this short-circuit at the very beginning of async handleMagicLinkKey and stop caring for it in the rest of the function?

Reply:
Because we want to exit after the polling has been cancelled or it has failed. There is no realistic case where the cancel button has been pressed before the polling has started. The point of this if is to return right before emittingErrors